proto/viam/app/data/v1/data.proto

```diff
@@ -173,3 +185,54 @@ message DeleteBinaryDataByIDsRequest {
 message DeleteBinaryDataByIDsResponse {
   uint64 deleted_count = 1;
 }
+
+// Annotation specifies the ways in which we can associate annotations with data, e.g. with a text tag or image bounding box
+message Annotation {
+  string tag = 1;
+}
+
+// AddAnnotationsToBinaryDataByFileIDsRequest requests adding annotations to the files when file IDs are provided
+message AddAnnotationsToBinaryDataByFileIDsRequest {
+  repeated string file_ids = 1;
+  repeated Annotation annotations = 2;
+}
+
+// AddAnnotationsToBinaryDataByFileIDsResponse returns IDs of added annotations in order to pass it to RemoveAnnotations
+message AddAnnotationsToBinaryDataByFileIDsResponse {
+  repeated string annotation_ids = 1;
+}
+
+// AddAnnotationsToBinaryDataByFilterRequest requests adding annotations to the files when a filter is provided
+message AddAnnotationsToBinaryDataByFilterRequest {
+  Filter filter = 1;
+  repeated Annotation annotations = 2;
+}
+
+// AddAnnotationsToBinaryDataByFilterResponse returns IDs of added annotations in order to pass it to RemoveAnnotations
+message AddAnnotationsToBinaryDataByFilterResponse {
+  repeated string annotation_ids = 1;
+}
+
+// RemoveAnnotationsFromBinaryDataByFilterRequest requests removing the annotations from the file when file IDs are provided
+message RemoveAnnotationsFromBinaryDataByFileIDsRequest {
+  repeated string file_ids = 1;
+  // If annotation_ids is empty, remove all annotations for the specified files.
+  repeated string annotation_ids = 2;
+}
+
+// RemoveAnnotationsFromBinaryDataByFilterResponse returns the number of binary files whose annotations were removed when file IDs are provided
+message RemoveAnnotationsFromBinaryDataByFileIDsResponse {
+  uint64 deleted_count = 1;
+}
+
+// RemoveAnnotationsFromBinaryDataByFilterRequest requests removing the annotations from the file when a filter is provided
+message RemoveAnnotationsFromBinaryDataByFilterRequest {
+  Filter filter = 1;
+  // If annotation_ids is empty, remove all annotations for the specified files.
```

Comment:
Based on feedback on another scope for something similar (passing in a DeleteRequest for a model and its repeated versions), the handling of this should be on the client, and the backend should only delete the given annotation_ids: [link](https://docs.google.com/document/d/1kKOhO_PzemaqXoHrP_F3KOYPJ3QMAo7dmNJdugxO5bI/edit?disco=AAAAjPsvUmU)

Reply:
Got it. Just took out the comment but lmk if I should do anything else to make note of this